### PR TITLE
amp-video-iframe: use InOb instead of ce.getIntersectionChangeEntry.

### PR DESCRIPTION
**Pull request by samouri**
Partial for https://github.com/ampproject/amphtml/issues/31540

Note: this is [a similar strategy](https://github.com/ampproject/amphtml/pull/31055#discussion_r520958803) being used by the new version of amp-video

### CODE REVIEW

File: extensions/amp-video-iframe/0.1/amp-video-iframe.js [hunk 1]
```diff
@@ -357,12 +357,27 @@ class AmpVideoIframe extends AMP.BaseElement {
...
 
   /**
+   * Creates an IntersectionObserver to fire a single intersection event.
+   *
    * @param {number} messageId
    * @private
    */
   postIntersection_(messageId) {
-    const {time, intersectionRatio} = this.element.getIntersectionChangeEntry();
+    const intersectionObserver = new IntersectionObserver((entries) => {
```
Comment (dvoytenko) [inline extensions/amp-video-iframe/0.1/amp-video-iframe.js:366]: How often is this invoked? Maybe it'd be better to just set a single instance of InOb on this instance of an element and then the callback will be more or less immediate? If this is more preferred, is this call throttled?

Comment (samouri) [reply extensions/amp-video-iframe/0.1/amp-video-iframe.js:366]: I followed the protocol on the newer thread for the new component: https://github.com/ampproject/amphtml/pull/31055#discussion_r520958803.

Its publisher-invoked and therefore I dont know how often it is called. If we end up merging a `getIntersection` or `measureIntersection => Promise` utility, we can use that

**Comment by samouri** [comment]
@dvoytenko: ready for another review now

File: extensions/amp-video-iframe/0.1/amp-video-iframe.js [hunk 2]
```diff
@@ -126,6 +127,12 @@ class AmpVideoIframe extends AMP.BaseElement {
      * @private
      */
     this.boundOnMessage_ = (e) => this.onMessage_(e);
+
+    /**
+     * @param {!Element} element
+     * @return {!Promise<IntersectionObserverEntry>} element
+     */
+    this.measureIntersection = (element) => measureIntersection(element);
```
Comment (dvoytenko) [inline extensions/amp-video-iframe/0.1/amp-video-iframe.js:135]: Sorry, but what's the value of this instance property?

Comment (samouri) [reply extensions/amp-video-iframe/0.1/amp-video-iframe.js:135]: Just for stubbing. Is it possible to stub a module using sinon?

Comment (dvoytenko) [reply extensions/amp-video-iframe/0.1/amp-video-iframe.js:135]: I see. This is a bit annoying. Maybe we can just stub the whole `IntersectionObserver` for this? If not, then I think we should at least make it a package-private or private member.

File: extensions/amp-video-iframe/0.1/amp-video-iframe.js [hunk 3]
```diff
@@ -275,6 +282,7 @@ class AmpVideoIframe extends AMP.BaseElement {
 
   /**
    * @param {!Event} event
+   * @return {Promise}
```
Comment (dvoytenko) [inline extensions/amp-video-iframe/0.1/amp-video-iframe.js:285]: This looks like a `{!Promise|undefined}` return type here.

File: extensions/amp-video-iframe/0.1/amp-video-iframe.js (hunk 2 above)
Comment (samouri) [reply extensions/amp-video-iframe/0.1/amp-video-iframe.js:135]: I'll stub in this case, but as a general pattern it feels too leaky (and laborious) for each client of `measureIntersection` to do that? Ideally they could just stub the function (the way Services would have allowed)

**Comment by samouri** [comment]
@dvoytenko PTAL

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**